services/ui-src/src/components/fields/ChoiceListField.tsx

```diff
@@ -44,14 +44,14 @@ export const ChoiceListField = ({
   const shouldDisableChildFields = !!props?.disabled;
 
   // set initial display value to form state field value or hydration value
-  const hydrationValue = props?.hydrate;
+  const hydrationValue = props?.hydrate || defaultValue;
```

Comment:
Note: This is the central idea behind the easier check to stop the initial "every time you blur it triggers" thing. Basically, if `props?.hydrate` is undefined, i.e. the field has never been saved before, set hydrationValue to defaultValue. This way we don't need the more complex checking logic for most field types (except for dynamic)

Reply:
_same change across multiple fields_

Reply:
Gave this a run. For whatever reason this is fine when I try it in Datefield, DropdownField, Numberfield and TextField but Choicelist HARD breaks with this change. The page will freeze up and become unresponsive. I suspect the useEffect gets triggered over and over again?